src/Hl7.Fhir.Base/ElementModel/TypedElementParseExtensions.cs

```diff
@@ -79,11 +79,11 @@ public static Model.Quantity ParseQuantity(this ITypedElement instance)
             return newQuantity;
         }
 
-        public static T ParsePrimitive<T>(this ITypedElement instance) where T : PrimitiveType, new()
+        public static T ParsePrimitive<T, U>(this U instance) where T : PrimitiveType, new() where U : IBaseElementNavigator<U>
```

Comment:
I think we need to provide an overload with the old signature + this one, since I think this is a breaking change (you had to change your code above, after all!).

Reply:
Every parse extension has now an extra method for IScopedNode, so we don't break the old signature.